runtime/vm/initsendtarget.cpp

```diff
@@ -225,16 +226,48 @@ initializeMethodRunAddressNoHook(J9JavaVM* vm, J9Method *method)
 	method->methodRunAddress = J9_BCLOOP_ENCODE_SEND_TARGET(J9_BCLOOP_SEND_TARGET_NON_SYNC);
 }
 
-J9Method cInitialStaticMethod = { 0, 0, J9_BCLOOP_ENCODE_SEND_TARGET(J9_BCLOOP_SEND_TARGET_INITIAL_STATIC), 0 };
-J9Method cInitialSpecialMethod = { 0, 0, J9_BCLOOP_ENCODE_SEND_TARGET(J9_BCLOOP_SEND_TARGET_INITIAL_SPECIAL), 0 };
-J9Method cInitialVirtualMethod = { 0, 0, J9_BCLOOP_ENCODE_SEND_TARGET(J9_BCLOOP_SEND_TARGET_INITIAL_VIRTUAL), 0 };
-#if defined(J9VM_OPT_VALHALLA_NESTMATES)
-J9Method cInvokePrivateMethod  = { 0, 0, J9_BCLOOP_ENCODE_SEND_TARGET(J9_BCLOOP_SEND_TARGET_INVOKE_PRIVATE), 0 };
-#endif /* J9VM_OPT_VALHALLA_NESTMATES */
-
 void
 initializeInitialMethods(J9JavaVM *vm)
 {
+	J9Method *cInitialStaticMethod = NULL;
+	J9Method *cInitialSpecialMethod = NULL;
+	J9Method *cInitialVirtualMethod = NULL;
+	J9Method *cInvokePrivateMethod = NULL;
+
+	if (IS_WARM_RUN(vm)) {
+		set_initial_methods(vm, &cInitialStaticMethod, &cInitialSpecialMethod, &cInitialVirtualMethod, &cInvokePrivateMethod);
+	} else {
+		if (IS_COLD_RUN(vm)) {
+			JVMIMAGEPORT_ACCESS_FROM_JAVAVM(vm);
+			cInitialStaticMethod = (J9Method *)imem_allocate_memory(sizeof(J9Method), J9MEM_CATEGORY_CLASSES);
```

Comment:
this just reminded me. We need a todo for catching mem alloc failures for our imem_allocates.